File: dart/scrabble-score/lib/scrabble_score.dart
int score([final String word = '']) {
  int sum = 0;
  for ( int i=0; i < word.length; i ++) {
    switch(word[i].toLowerCase()) {
      case 'a':
      case 'e':
      case 'i':
      case 'l':
      case 'n':
      case 'o':
      case 'r':
      case 's':
      case 't':
      case 'u':
        sum += 1;
        break;
      case 'd':
      case 'g':
        sum += 2;
        break;
      case 'b':
      case 'c':
      case 'm':
      case 'p':
        sum += 3;
        break;
      case 'f':
      case 'h':
      case 'v':
      case 'w':
      case 'y':
        sum += 4;
        break;
      case 'k':
        sum += 5;
        break;
      case 'j':
      case 'x':
        sum += 8;
        break;
      case 'q':
      case 'z':
        sum += 10;
        break;
      default:
        sum += 0;
    }
  }
  return sum;
}

File: php/scrabble-score/scrabble-score.php
<?php

/**
 * Calculate the scrabble score for given word.
 *
 * @param string $word The string to calculate to score for.
 *
 * @return int The calculated scrabble score as integer.
 */
function score(string $word = ""): int
{
  $sum = 0;

  for ($idx = 0; $idx < strlen($word); $idx++) {
    $char = substr($word, $idx, 1);
    switch (true) {
      case preg_match("/[aeilnorstu]/i", $char):
        $sum += 1;
        break;
      case preg_match("/[dg]/i", $char):
        $sum += 2;
        break;
      case preg_match("/[bcmp]/i", $char):
        $sum += 3;
        break;
      case preg_match("/[fhvwy]/i", $char):
        $sum += 4;
        break;
      case preg_match("/[k]/i", $char):
        $sum += 5;
        break;
      case preg_match("/[jx]/i", $char):
        $sum += 8;
        break;
      case preg_match("/[qz]/i", $char):
        $sum += 10;
        break;
      default:
        $sum += 0;
    }
  }

  return $sum;
}
